### PR TITLE
[CI] Upgrade CI

### PR DESCRIPTION
**Pull request by yongwww**
Component | Current Version | Upgraded Version
-- | -- | --
Docker of ci_cpu, gpu, arm, etc. | 20.04 | 22.04
Docker of ci_i386 | 18.04 | 20.04 (No 22.04 so far)
CMake | 3.18 | 3.20.0
LLVM |  - | 13, 14, 15, 16
ROCM | 4.3 | 5.3 (4.3 has unmet packages in upgraded container)
CUDA | 11.7 | 11.8
Python | 3.7 | 3.8
Pylint | 2.4.4 | 2.17.2
Torch | - | 2.0
torchvision | - | 0.15.1
libtorch | - | 2.0
Golang| 1.12 |  1.18
emsdk| 2.0.7| 3.1.30
Verilator| 4.104 | 5.002
Numpy| 1.21 | 1.23
DGL| 0.7.2 | 1.0.0
clang-format | 10 | 15

### CODE REVIEW

**Comment by tqchen** [comment]
cc @masahi

File: docker/Dockerfile.ci_lint
Comment (driazati) [inline docker/Dockerfile.ci_lint:29]:
```suggestion
ENV TVM_VENV /venv/apache-tvm-py3.8
ENV DEBIAN_FRONTEND noninteractive
```

we probably also need this to fix the timeout here: https://ci.tlcpack.ai/blue/organizations/jenkins/tvm-docker/detail/PR-14635/6/pipeline/77#step-212-log-1473

Comment (yongwww) [reply docker/Dockerfile.ci_lint:29]: Thanks, got a fix with
``` 
RUN bash /install/ubuntu_setup_tz.sh
```

**Review by leandron — CHANGES_REQUESTED** [review]
Hi @yongwww, thanks for the PR. I see this is marked as work in progress.

Can I just ask the we validate the images in a full CI round before merging this PR, that not only they build the images correctly, but also that tests pass with the resultant images, as you're doing with https://github.com/apache/tvm/pull/14651. These are a lot of changes for one PR, but I appreciate that moving versions is hard.

I'm marking this as "request changes" so that we don't accidentally merge this based on the local CI results.

**Comment by yongwww** [comment]
Thanks, @leandron ! I agree that this pull request should be held off until #14651 turns green. I've marked it as a work-in-progress and will revisit it

**Comment by yongwww** [comment]
@tvm-bot  rerun

**Comment by yongwww** [comment]
@Aleksei-grovety  great observation! I rolled back the TensorFlow upgrade because the linking process failed for the TFLite runtime during the TVM build (such as lots of errors like "multiple definition ...", and "undefined reference to..."). This issue needs specific update in the tflite runtime build, which I haven't had time to address. In fact, I think we should consider using TFLite runtime Python package (https://pypi.org/project/tflite-runtime/) provided by Google.

**Comment by yongwww** [comment]
@tvm-bot rerun

**Comment by masahi** [comment]
@yongwww Feel free to drop PT 2.0 update from this PR. I'm aware that some PT frontend tests are broken with 2.0. I can continue my PR to fix them, once we get Python 3.10 from here.

**Comment by yongwww** [comment]
@masahi I have fixed some of the failed pt tests in https://github.com/apache/tvm/pull/14651, and skip those tests if there is new unsupported op.

**Comment by yongwww** [comment]
@tvm-bot rerun

**Comment by yongwww** [comment]
@tvm-bot rerun

**Comment by yongwww** [comment]
@tvm-bot rerun

**Comment by yongwww** [comment]
@tvm-bot  rerun

**Comment by yongwww** [comment]
@tvm-bot rerun

**Comment by yongwww** [comment]
@tvm-bot  rerun

**Comment by yongwww** [comment]
@tvm-bot  rerun

**Comment by yongwww** [comment]
@tvm-bot  rerun

**Comment by yongwww** [comment]
The PR is ready for review, pls take a look. cc: @masahi @driazati @leandron @tqchen

**Review by driazati — APPROVED** [review]
Great work! Thanks @yongwww, the changes LGTM if the build from #14651 passes

**Comment by yongwww** [comment]
@tvm-bot rerun

**Comment by yongwww** [comment]
@tvm-bot rerun

**Comment by yongwww** [comment]
CI #14651 is green

**Review by tqchen — APPROVED** [review]

**Comment by tqchen** [comment]
Thank you @yongwww for heroic effort!

**Comment by psrivas2** [comment]
cc @leandron can you take another look at this PR. I think this is ready to be merged.

**Comment by tqchen** [comment]
ping @leandron

**Comment by yongwww** [comment]
@tvm-bot  rerun

File: docker/install/ubuntu_install_ethosn_driver_stack.sh
Comment (leandron) [inline docker/install/ubuntu_install_ethosn_driver_stack.sh:51]: Do you mind opening an issue to report what is the error that makes this script now need `werror=False`?

I understand it will have some output, but can you add that output in the issue, so that somebody can pick that up to investigate?

File: docker/Dockerfile.ci_arm
Comment (leandron) [inline docker/Dockerfile.ci_arm:54]: Given we will be installing LLVM 15 from apt repositories, this can be removed in future. We don't need to block this PR, but can you open an issue so that we can actually do this in a follow-up PR?

**Review by leandron — COMMENTED** [review]
I'm happy to merge this once https://github.com/apache/tvm/pull/14651 CI is green. At the moment it is running, as the GitHub actions needed to be approved, which I just did.

I marked a few action I feel we need to take as follow-up PRs, can you just open those as issues so that we can pick them up asap. There no need to block this PR because the actual changes, but can you open the issues, please?

File: docker/install/ubuntu_install_ethosn_driver_stack.sh
Comment (yongwww) [reply docker/install/ubuntu_install_ethosn_driver_stack.sh:51]: I got error like "error: loop variable 'pair' creates a copy from type 'const std::pair<const unsigned int, std::pair<long unsigned int, long unsigned int> >' [-Werror=range-loop-construct] for (const auto pair : passAgentIdRanges)", it should be a warning, so I added werror=False in scons install for that, I think the ideal solution is to modify the code in https://github.com/Arm-software/ethos-n-driver-stack to fix this warning. More details about the error info. please refer to the log https://ci.tlcpack.ai/blue/rest/organizations/jenkins/pipelines/tvm-docker/branches/PR-14635/runs/5/nodes/73/steps/129/log/?start=0

File: docker/Dockerfile.ci_arm
Comment (yongwww) [reply docker/Dockerfile.ci_arm:54]: https://github.com/apache/tvm/issues/14758 issue was created for this

**Comment by tqchen** [comment]
Going to merge this first in 24 hours so we can proceed with the updates on the other side

File: docker/Dockerfile.ci_arm
Comment (leandron) [reply docker/Dockerfile.ci_arm:54]: Thanks!

**Comment by tqchen** [comment]
Thank you @yongwww for heroic effort

**Pull request merged**
